src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MIN_LEG_LENGTH = 1;
+    private static final int MAX_LEG_LENGTH = 100;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureTypes type = FigureTypes.values()[random.nextInt(FigureTypes.values().length)];
+        switch (type) {
+            case SQUARE:
+                return new Square(random.nextInt(MIN_LEG_LENGTH,MAX_LEG_LENGTH),
```

Comment:
format your file - Ctrl Alt L, space after comma
random.nextInt(MIN_LEG_LENGTH,MAX_LEG_LENGTH) - move this repeated logic to a separate private utility method